package.json

```diff
@@ -2,7 +2,7 @@
   "name": "github-pages-deploy-action",
   "description": "GitHub action for building a project and deploying it to GitHub pages.",
   "version": "3.2.4",
-  "main": "lib/main.js",
+  "main": "lib/lib.ts",
```

Comment:
This needs to be javascript, with type definitions. See ["Publishing" — Typescript](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) for more information.
```suggestion
  "main": "lib/lib.js",
  "types": "./lib/lib.d.ts",
```

Reply:
Added the type declaration to the build script